packages/beacon-node/src/chain/factory/block/body.ts

```diff
@@ -175,28 +162,45 @@ export async function prepareExecutionPayload(
   finalizedBlockHash: RootHex,
   state: CachedBeaconStateBellatrix,
   suggestedFeeRecipient: string
-): Promise<PayloadId | null> {
-  const parentHash = extractExecutionPayloadParentHash(chain, state);
-  if (parentHash === null) return null;
+): Promise<bellatrix.ExecutionPayload> {
+  const parentHashRes = getExecutionPayloadParentHash(chain, state);
+  if (parentHashRes.isPremerge) {
+    // Before finding the merge block, put empty ExecutionPayload to signal merge block not found
+    return ssz.bellatrix.ExecutionPayload.defaultValue();
+  }
 
+  const {parentHash} = parentHashRes;
   const timestamp = computeTimeAtSlot(chain.config, state.slot, state.genesisTime);
   const prevRandao = getRandaoMix(state, state.epochCtx.epoch);
 
-  const payloadId =
-    chain.executionEngine.payloadIdCache.get({
-      headBlockHash: toHex(parentHash),
-      finalizedBlockHash,
-      timestamp: numToQuantity(timestamp),
-      prevRandao: toHex(prevRandao),
-      suggestedFeeRecipient,
-    }) ??
-    (await chain.executionEngine.notifyForkchoiceUpdate(parentHash, safeBlockHash, finalizedBlockHash, {
-      timestamp,
-      prevRandao,
-      suggestedFeeRecipient,
-    }));
-  if (!payloadId) throw new Error("InvalidPayloadId: Null");
-  return payloadId;
+  const payloadIdCached = chain.executionEngine.payloadIdCache.get({
+    headBlockHash: toHex(parentHash),
+    finalizedBlockHash,
+    timestamp: numToQuantity(timestamp),
+    prevRandao: toHex(prevRandao),
+    suggestedFeeRecipient,
+  });
+
+  if (payloadIdCached !== undefined) {
+    return await chain.executionEngine.getPayload(payloadIdCached);
+  }
+
+  // prepareExecutionPayload will throw error via notifyForkchoiceUpdate if
+  // the EL returns Syncing on this request to prepare a payload
+  // TODO: Handle only this case, DO NOT put a generic try / catch that discards all errors
+  const payloadId = await chain.executionEngine.notifyForkchoiceUpdate(parentHash, safeBlockHash, finalizedBlockHash, {
+    timestamp,
+    prevRandao,
+    suggestedFeeRecipient,
+  });
```

Comment:
@g11tech please re-add the try / catch here only and ignore only a range of expected errors, always logging to some appropriate logging level, else throw